chore: update migration docs for EDOT Node.js (#431)
elastic/opentelemetry: docs/reference/edot-sdks/nodejs/migration.md

For example: `OTEL_EXPORTER_OTLP_HEADERS=foo=bar,baz=quux`.

### `centralConfig`

The Elastic APM Node.js agent [`centralConfig`](apm-agent-nodejs://reference/configuration.md#central-config) option corresponds to the EDOT Node.js [`ELASTIC_OTEL_OPAMP_ENDPOINT`](/reference/edot-sdks/nodejs/configuration.md#turn-on-central-configuration) option.

For example: `export ELASTIC_OTEL_OPAMP_ENDPOINT=http://localhost:4320/v1/opamp`.

:::{warning}

> To activate central configuration for EDOT Node.js is necessary to enable it by following the [configuration guide](/reference/central-configuration.md). Also there is a difference in which options can be configured. You can find a list in [central configuration settings](/reference/edot-sdks/nodejs/configuration.md#central-configuration-settings).

Reviewer: ```suggestion
To use central configuration for EDOT Node.js it is necessary to enable it by following the [configuration guide](/reference/central-configuration.md). Also there is a difference in which options can be configured. You can find a list in [central configuration settings](/reference/edot-sdks/nodejs/configuration.md#central-configuration-settings).
```

:::

### `cloudProvider`

> The Elastic APM Node.js agent [`cloudProvider`](apm-agent-nodejs://reference/configuration.md#cloud-provider) option does not corresponds directly to any OpenTelemetry option but you can get similar behavior by properly setting [`OTEL_NODE_RESOURCE_DETECTORS`](https://opentelemetry.io/docs/zero-code/js/configuration/#sdk-resource-detector-configuration) option. If you set this option make sure you add along with the cloud detector the non-cloud detectors that apply to your service. For a full list of detectors check [OTEL_NODE_RESOURCE_DETECTORS details](/reference/edot-sdks/nodejs/configuration.md#otel_node_resource_detectors-details). Not setting this option is the equivalent of `auto`.

Reviewer: ```suggestion
The Elastic APM Node.js agent [`cloudProvider`](apm-agent-nodejs://reference/configuration.md#cloud-provider) option does not correspond directly to any OpenTelemetry option but you can get similar behavior by properly setting [`OTEL_NODE_RESOURCE_DETECTORS`](https://opentelemetry.io/docs/zero-code/js/configuration/#sdk-resource-detector-configuration) option. If you set this option make sure you add along with the cloud detector the non-cloud detectors that apply to your service. For a full list of detectors check [OTEL_NODE_RESOURCE_DETECTORS details](/reference/edot-sdks/nodejs/configuration.md#otel_node_resource_detectors-details). Not setting this option is the equivalent of `auto`.
```

For example: `OTEL_NODE_RESOURCE_DETECTORS=os,env,host,serviceinstance,process,aws` will make the agent query for AWS metadata only and use other non-cloud detectors to enrich that metadata.

> ### `contextPropagationOnly`

Author: note for reviewer: I just moved this option here to keep the alphabetical order. It was already added in https://github.com/elastic/opentelemetry/pull/412

    - [`ELASTIC_OTEL_HOST_METRICS_DISABLED=true`](/reference/edot-sdks/nodejs/configuration.md#elastic_otel_host_metrics_disabled-details) to remove the overhead from metrics collection by the `@opentelemetry/host-metrics` package.
    - [`OTEL_NODE_DISABLED_INSTRUMENTATIONS=runtime-node`](/reference/edot-sdks/nodejs/configuration.md#otel_node_disabledenabled_instrumentations-details) to remove metrics collection overhead from the `@opentelemetry/instrumentation-runtime-node` instrumentation

For example:

```
ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY=true
OTEL_METRICS_EXPORTER=none
ELASTIC_OTEL_HOST_METRICS_DISABLED=true
OTEL_NODE_DISABLED_INSTRUMENTATIONS=runtime-node
```

### `disableInstrumentations`

The Elastic APM Node.js agent [`disableInstrumentations`](apm-agent-nodejs://reference/configuration.md#disable-instrumentations) option corresponds to the EDOT Node.js [`OTEL_NODE_DISABLED_INSTRUMENTATIONS`](/reference/edot-sdks/nodejs/configuration.md#otel_node_disabledenabled_instrumentations-details) option.

For example: `OTEL_NODE_DISABLED_INSTRUMENTATIONS=express,mysql`.

### `disableMetrics`

> The Elastic APM Node.js agent [`disableMetrics`](apm-agent-nodejs://reference/configuration.md#disable-metrics) option corresponds to the EDOT Node.js [`OTEL_METRICS_EXPORTER`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection) option. Set it to none to disable the collection of metrics.

Reviewer: This is a little off.  `disableMetrics` allows specifying a set of metric name patterns to exclude. So `OTEL_METRICS_EXPORTER=none` is equivalent to `disableMetrics: '*'`.  

To get something similar to `disableMetrics: 'nodejs.*,mycounter'` could be achieved using *Views*. However, that requires us (me? I'd volunteered) writing the "advanced setup" docs that shows to write bootstrap code (i.e. as minimally documented at https://github.com/elastic/elastic-otel-node/blob/main/examples/telemetry.mjs).

As well, it would perhaps be helpful to users to mention some other things:
- `nodejs.*` and `v8js.*` metrics are all created by the `@opentelemetry/instrumentation-runtime-node` instrumentation and can be disabled via `OTEL_NODE_DISABLED_INSTRUMENTATIONS=runtime-node`.
- `process.*` (and `system.*`, if overriding the default views used by the EDOT Node.js SDK) metrics are generated by the `@opentelemetry/host-metrics` package and can be fully disabled via `ELASTIC_HOST_OTEL_METRICS_DISABLED=true`

For example: `OTEL_METRICS_EXPORTER=none`.

### `disableSend`

> The Elastic APM Node.js agent [`disableSend`](apm-agent-nodejs://reference/configuration.md#disable-send) does not corresponds to any Opentelemetry option. You can get the same behavior by setting to the value `none` on the following environment vars dedicated to [exporter selection](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection). Note that you have control per signal.

Reviewer: ```suggestion
The Elastic APM Node.js agent [`disableSend`](apm-agent-nodejs://reference/configuration.md#disable-send) does not corresponds to a single OpenTelemetry option. You can get the same behavior by setting to the value `none` on the following environment vars dedicated to [exporter selection](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection). Note that you have control per signal.
```

- `OTEL_TRACES_EXPORTER` for traces.
- `OTEL_METRICS_EXPORTER` for metrics.
- `OTEL_LOGS_EXPORTER` for logs.

For example `OTEL_TRACES_EXPORTER=none OTEL_METRICS_EXPORTER=none OTEL_LOGS_EXPORTER=none` will completelly disable sending but `OTEL_TRACES_EXPORTER=none` will disable only the sending of traces.

### `environment`

> The Elastic APM Node.js agent [`environment`](apm-agent-nodejs://reference/configuration.md#environment) option corresponds to setting the `deployment.environment` key in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes).

Reviewer: nit: Can we update this to the current `deployment.environment.name`?  `deployment.environment` was deprecated relatively recently.  This could be done in a separate change.

### `globalLabels`

The Elastic APM Node.js agent [`globalLabels`](apm-agent-nodejs://reference/configuration.md#global-labels) option corresponds to adding `key=value` comma separated pairs in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes).

For example: `OTEL_RESOURCE_ATTRIBUTES=alice=first,bob=second`. Such labels will result in labels.key=value attributes on the server. For example, `labels.alice=first`.

### `hostname`

The Elastic APM Node.js agent [`hostname`](apm-agent-nodejs://reference/configuration.md#hostname) option corresponds to setting the `host.name` key in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes).

For example: `OTEL_RESOURCE_ATTRIBUTES=host.name=myhost`.

### `instrument`

The Elastic APM Node.js agent [`instrument`](apm-agent-nodejs://reference/configuration.md#instrument) option can be achieved with [`OTEL_NODE_ENABLED_INSTRUMENTATIONS`](/reference/edot-sdks/nodejs/configuration.md#otel_node_disabledenabled_instrumentations-details) option.

> For example: `OTEL_NODE_ENABLED_INSTRUMENTATIONS=none` will make EDOT Node.js .

Reviewer 2: @david-luna What should we add here?

For example: `OTEL_RESOURCE_ATTRIBUTES=alice=first,bob=second`. Such labels will result in labels.key=value attributes on the server. For example, `labels.alice=first`.

### `hostname`

> The Elastic APM Node.js agent [`environment`](apm-agent-nodejs://reference/configuration.md#instrument) option corresponds to setting the `host.name` key in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes).

Reviewer: ```suggestion
The Elastic APM Node.js agent [`hostname`](apm-agent-nodejs://reference/configuration.md#hostname) option corresponds to setting the `host.name` key in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes).
```

For example: `OTEL_RESOURCE_ATTRIBUTES=host.name=myhost`.

### `instrument`

> The Elastic APM Node.js agent [`environment`](apm-agent-nodejs://reference/configuration.md#instrument) option corresponds to [`OTEL_NODE_ENABLED_INSTRUMENTATIONS`](/reference/edot-sdks/nodejs/configuration.md#otel_node_disabledenabled_instrumentations-details) option.

Reviewer: ```suggestion
The Elastic APM Node.js agent [`instrument`](apm-agent-nodejs://reference/configuration.md#instrument) option corresponds to [`OTEL_NODE_ENABLED_INSTRUMENTATIONS`](/reference/edot-sdks/nodejs/configuration.md#otel_node_disabledenabled_instrumentations-details) option.
```

For example: `OTEL_NODE_ENABLED_INSTRUMENTATIONS=none`.

> TODO: this works but it shows a warning. Should we consider this as good enough???

Author: note for reviewer: discuss this comment here

Reviewer: Perhaps change the first paragraph to say something like:

> ... the equivalent of the `instrument: false` option can be achieved with `OTEL_NODE_ENABLED_INSTRUMENTATIONS=none` ...

rather than saying it "corresponds to".

Reviewer: Ah, `OTEL_NODE_ENABLED_INSTRUMENTATIONS=none` is a hack -- `none` isn't an actual blessed/supported value -- which is why the warning log message.  I think we should mention that this will result in a warning message.

If we want to, we could mention that the equivalent when using code bootstrapping is to explicitly pass in an empty array to the `instrumentations: []` option to `startNodeSDK(...)`. (Again, we don't currently have a doc landing page for that. We could hide this here inside an HTML comment for now or just add it later.)

| `trace`               | `verbose`      |
| `trace`               | `all`          |

### `longFieldMaxLength`

> The Elastic APM Node.js agent [`longFieldMaxLength`](apm-agent-nodejs://reference/configuration.md#long-field-max-length) option does not corresponds directly to an OpenTelemetry option but you can get similar behavior by properly setting [`OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#attribute-limits) option. This limit can also be set for specific signals

Reviewer: ```suggestion
The Elastic APM Node.js agent [`longFieldMaxLength`](apm-agent-nodejs://reference/configuration.md#long-field-max-length) option does not corresponds directly to an OpenTelemetry option but you can get similar behavior by properly setting [`OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#attribute-limits) option. This limit can also be set for specific signals.
```

- `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT` for traces.
- `OTEL_LOGRECORD_ATTRIBUTE_VALUE_LENGTH_LIMIT` for logs.

> For example: `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT=1000` will limit all attributes to a thousad chars long.

Reviewer: ```suggestion
For example: `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT=1000` will limit attribute string values to a thousand characters.
```

For example: `OTEL_METRIC_EXPORT_INTERVAL=30000`.

### `opentelemetryBridgeEnabled`

If you were using the Elastic APM Node.js agent [`opentelemetryBridgeEnabled`](apm-agent-nodejs://reference/configuration.md#opentelemetry-bridge-enabled) option you can drop it safely and keep using `@openetelemetry/api` to manually instrument your service.

> 

Reviewer: I'm torn. Perhaps just don't mention this?  The OTel Bridge was never GA. I'm fine either way.

You can also set the service name using [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes). For example: `OTEL_RESOURCE_ATTRIBUTES=service.name=myservice`. A value in `OTEL_SERVICE_NAME` takes precedence over a `service.name` value in `OTEL_RESOURCE_ATTRIBUTES`.

### `serviceNodeName`

> The Elastic APM Node.js agent [`serviceNodeName`](apm-agent-nodejs://reference/configuration.md#service-node-name) option corresponds to setting the `service.instance.id` key in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes). Warning: by default this is a generated unique ID; if you set this it must be a unique value for each service instance otherwise metric views cannot be correctly aggregated nor disambiguated

Reviewer: ```suggestion
The Elastic APM Node.js agent [`serviceNodeName`](apm-agent-nodejs://reference/configuration.md#service-node-name) option corresponds to setting the `service.instance.id` key in [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_resource_attributes). Warning: by default this is a generated unique ID; if you set this it must be a unique value for each service instance otherwise metric views cannot be correctly aggregated or disambiguated.
```
